### PR TITLE
Use python@3.8 on macOS

### PR DESCRIPTION
**Pull request by jamiesnape**
Needs macOS CI update.

----
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/13031)

### CODE REVIEW

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-mojave-unprovisioned-clang-bazel-experimental-release please

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-mojave-unprovisioned-clang-bazel-experimental-release please

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-mojave-unprovisioned-clang-bazel-experimental-release please

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-catalina-unprovisioned-clang-cmake-experimental-release please
@drake-jenkins-bot mac-mojave-unprovisioned-clang-bazel-experimental-release please

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-catalina-unprovisioned-clang-cmake-experimental-release please
@drake-jenkins-bot mac-mojave-unprovisioned-clang-bazel-experimental-release please

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-catalina-unprovisioned-clang-cmake-experimental-release please
@drake-jenkins-bot mac-mojave-unprovisioned-clang-bazel-experimental-release please

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-catalina-unprovisioned-clang-cmake-experimental-release please
@drake-jenkins-bot mac-mojave-unprovisioned-clang-bazel-experimental-release please

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-catalina-unprovisioned-clang-bazel-experimental-release please
@drake-jenkins-bot mac-mojave-unprovisioned-clang-cmake-experimental-release please

**Comment by jamiesnape** [comment]
@drake-jenkins-bot mac-catalina-unprovisioned-clang-bazel-experimental-release please
@drake-jenkins-bot mac-mojave-unprovisioned-clang-bazel-experimental-release please

**Comment by jamiesnape** [comment]
+@jwnimmer-tri and +@EricCousineau-TRI for review.

There is certainly going to be a cleanup PR following this (not least because drake-visualizer still needs fixing).

**Comment by EricCousineau-TRI** [comment]
___
*[tools/workspace/pybind11/test/sample_header_documentation_pybind_expected_python_3.8.xml, line 1 at r1](https://reviewable.io/reviews/robotlocomotion/drake/13031#-M4tSg5u2Bb6YDxh_99a:-M4tSg5u2Bb6YDxh_99b:btl5vgv) ([raw file](https://github.com/robotlocomotion/drake/blob/6d6dca138f87a48a9dd3b65b642aedfc8a580ebe/tools/workspace/pybind11/test/sample_header_documentation_pybind_expected_python_3.8.xml#L1)):*
> ```XML
> <?xml version="1.0" ?>
> ```

This file is marked as executable.
Can you remove that bit?

**Comment by EricCousineau-TRI** [comment]
Just so I (and others) know, what is the `do not merge` in place for?

**Comment by jamiesnape** [comment]
Needs a CI update.

File: doc/python_bindings.rst [hunk 1]
```diff
@@ -16,6 +16,11 @@ installed as a single package called ``pydrake``.
    Please uninstall Anaconda or remove the Anaconda ``bin`` directory from the
    ``PATH`` before building or using the Drake Python bindings.
 
+.. warning::
+   On macOS, Drake only supports Python 3.8, which is located at
+   ``/usr/local/opt/python@3/bin/python3`` and is not usually on the
```
Comment (RussTedrake) [inline doc/python_bindings.rst:21]: This was not correct for me.  I needed `/usr/local/opt/python@3.8/bin`
@jwnimmer-tri  -- This was vicious enough that it probably needs to be very visible in the release notes.

Comment (jwnimmer-tri) [reply doc/python_bindings.rst:21]: We have it as the very first announcement: https://github.com/jwnimmer-tri/drake/blob/release_notes-v0.17.0/doc/release_notes/v0.17.0.rst#announcements.  Can you comment on #13020 review where you'd like to see more?

**Pull request merged**